From 9a96777728c3aa8901af36b8ddeeb89522673ac7 Mon Sep 17 00:00:00 2001
From: Double Lo <dblo@cypress.com>
Date: Tue, 5 Jun 2018 21:39:58 -0500
Subject: [PATCH 4705/5242] MLK-19517-13 brcmfmac: disable command decode in
 sdio_aos for 4354

commit  2565626257e89ebb5fece66150dd583bc0bd8b5f from
https://source.codeaurora.org/external/imx/linux-imx.git

Transaction between AOS and SDIOD is not protected, and if cmd 52
received in AOS and in the middle of response state changed from AOS to
SDIOD, response is corrupted and it causes to SDIO Host controller to
hang.

Signed-off-by: Lo-Hsiang Lo <double.lo@cypress.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../wireless/broadcom/brcm80211/brcmfmac/sdio.c    |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
index c002efe..739c0b6 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
@@ -3522,6 +3522,7 @@ static void brcmf_sdio_sr_init(struct brcmf_sdio *bus)
 
 	if (bus->ci->chip == CY_CC_43012_CHIP_ID ||
 	    bus->ci->chip == BRCM_CC_4339_CHIP_ID ||
+	    bus->ci->chip == BRCM_CC_4354_CHIP_ID ||
 	    bus->ci->chip == BRCM_CC_4345_CHIP_ID) {
 		cardcap = SDIO_CCCR_BRCM_CARDCAP_CMD_NODEC;
 	} else {
-- 
1.7.9.5

